File: 3973-flip-square-submatrix-vertically/flip-square-submatrix-vertically.py
class Solution:
    def reverseSubmatrix(self, grid: List[List[int]], x: int, y: int, k: int) -> List[List[int]]:
        arr = [row[y:y+k] for row in grid[x:x+k]][::-1]
        row=0
        for i in range(x,x+k):
            col=0
            for j in range(y,y+k):
                grid[i][j] = arr[row][col]
                col+=1
            row+=1

        return grid

        

File: 3973-flip-square-submatrix-vertically/flip-square-submatrix-vertically.java
class Solution {
    public int[][] reverseSubmatrix(int[][] grid, int x, int y, int k) {

        int[][] arr = new int[k][k];
        int row = 0;
        for (int i = x ; i < x+k ;i++){
            int col =0;
            for (int j = y ; j< y+k ; j++){
                arr[row][col]=grid[i][j];
                col++;
            }
            row++;
        }

        row = k-1;
        for (int i = x ; i < x+k ;i++){
            int col =0;
            for (int j = y ; j< y+k ; j++){
                grid[i][j]=arr[row][col];
                col++;
            }
            row--;
        }

        return grid;

        
    }
}